keyboards/planck/keymaps/purple-rw/keymap.c

```diff
@@ -0,0 +1,548 @@
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |      |      |      |      |      |      |      |      |      |      |      |      |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * | Exit |      |      | 1 #  | c A- | v O- | n -E | m -U | 1 #  |      |      |      |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_PLOVER] = LAYOUT_planck_grid(
+    XXXXXXX, KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC,
+    XXXXXXX, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,
+    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+    EXT_PLV, XXXXXXX, XXXXXXX, KC_1,    KC_C,    KC_V,    KC_N,    KC_M,    KC_1,    XXXXXXX, XXXXXXX, XXXXXXX
+),
+
+/* Adjust
+ * ,-----------------------------------------------------------------------------------.
+ * | Exit |Brght+| Sat+ | Hue+ |RGBmod| BL+  |      |      |      |      |      |Debug |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |      |Brght-| Sat- | Hue- | RGB  | BL-  |AGnorm|Colemk|Qwerty|HndsDn|Semimk|Plover|
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |      |      |      |      |  BL  |BLbrth|AGswap|AudOn |AudOff|MusOn |MusNxt|      |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |      |      |      |      |      | Exit | Exit |      |      |      |Reset |BootL |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_ADJST] = LAYOUT_planck_grid(
+    ADJST,   RGB_VAI, RGB_SAI, RGB_HUI, RGB_MOD, BL_UP,   XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, DB_TOGG,
+    XXXXXXX, RGB_VAD, RGB_SAD, RGB_HUD, RGB_TOG, BL_DOWN, AG_NORM, COLEMAK, QWERTY,  HANDSDN, SEMIMAK, PLOVER,
+    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, BL_TOGG, BL_BRTG, AG_SWAP, AU_ON,   AU_OFF,  MU_ON,   MU_NEXT, XXXXXXX,
+    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, ADJST,   ADJST,   XXXXXXX, XXXXXXX, XXXXXXX, QK_RBT,  QK_BOOT
+)
+
+};
+
+#ifdef AUDIO_ENABLE
+    float plover_song[][2]     = SONG(PLOVER_SOUND);
+    float plover_gb_song[][2]  = SONG(PLOVER_GOODBYE_SOUND);
+#endif
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+  return update_tri_layer_state(state, _LOWER, _RAISE, _MODS);
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+#ifdef CONSOLE_ENABLE
+    if (record->event.pressed) {
+        uprintf("0x%04X,%u,%u,%u\n", keycode, record->event.key.row, record->event.key.col,
+                get_highest_layer(layer_state));
+    }
+#endif
+
+    switch (keycode) {
+        case COLEMAK:
+            if (record->event.pressed) {
+                set_single_persistent_default_layer(keycode);
```

Comment:
This looks wrong, the `keycode`'s value will be around `0x7F00`, definitely not a correct value for the layer number. You probably wanted `keycode - QK_USER` instead.

Also, i have no idea why you are disabling `_ADJST` on release, do you need it to be kept enabled while the custom keycode is held, for some reason? Otherwise, all the `layer_on`/`layer_off` could most likely be replaced by a 
single `layer_move(_COLEMAK)` on the `if (pressed)` block

At that point, if you:
- Keep all these base layers as the first ones in the layers' enum (or not the 1st but correlative)
- Define the custom keycodes in the same order as the layers

A single piece of code could handle all of them without duplication 
```c
case COLEMAK ... SEMINAK:
    if (record->event.pressed) {
        set_single_persistent_default_layer(keycode - QK_USER);
        layer_move(keycode - QK_USER);
    }
    return false;
```

Reply:
Cool.  Missed that layer_move() function from the doc.  Will try it put.

Reply:
Also, layer_move or other layer functions shouldn't be necessary here, as the default layer mask is handled separately, and merged with the layer mask for layer related stuff.  Eg, it's an extra step that isn't needed.